webpack/assets/javascripts/react_app/components/factCharts/index.js

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { Button, Modal } from 'react-bootstrap';
+import helpers from '../../common/helpers';
+import Chart from '../common/charts/Chart';
+import Loader from '../common/Loader';
+import MessageBox from '../common/MessageBox';
+import { getLargePieChartConfig, setTitle } from '../../../services/ChartService';
+import { STATUS } from '../../constants';
+import * as FactChartActions from '../../redux/actions/factCharts/';
+
+class FactChart extends React.Component {
+  constructor(props) {
+    super(props);
+    helpers.bindMethods(this, ['onClick', 'closeModal']);
+  }
+
+  onClick() {
+    const { showModal, getChartData, data: { id, path, title } } = this.props;
+
+    getChartData(path, id);
+    showModal(id, title);
+  }
+
+  closeModal() {
+    this.props.closeModal();
+  }
+
+  render() {
+    const { factChart, data: { id, title } } = this.props;
+
+    const showModal = factChart.displayModal[id];
```

Comment:
its the same name as the action function, shouldn't it has a unique name?
or maybe map `displayModal` to a prop?